torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -575,86 +581,213 @@ void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
     const auto &valsList = iVal.toBoolList();
     const auto &vals = c10::impl::toVector(valsList);
     if (vals.empty()) {
-      return;
+      return nullptr;
     }
     t.reset(glow::ElemKind::BoolTy, {vals.size()});
     auto handle = t.getHandle<bool>();
     for (size_t i = 0; i < vals.size(); ++i) {
       handle.raw(i) = vals[i];
     }
   } else if (iVal.isTensor()) {
-    assert(false && "not yet implemented tensor constants");
+    const at::Tensor &val = iVal.toTensor();
+    t = ptTensorToGlowTensor(val);
   } else if (iVal.isNone()) {
     // Nothing to do for None
-    return;
+    return nullptr;
   } else {
     assert(false && "Unsupported constant type");
   }
 
-  glow::Constant *glowConstant = mod->createConstant("constant", std::move(t));
+  return mod->createConstant("constant", std::move(t));
+}
+
+void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+  assert(inputs.size() == 0);
+  assert(outputs.size() == 1);
+  (void)inputs;
+
+  auto optionalIValue = torch::jit::toIValue(outputs[0]);
+  assert(optionalIValue.has_value() && "Constants should have IValue outputs");
+  const torch::jit::IValue iVal = *optionalIValue;
+
+  glow::Constant *glowConstant = glowConstantFromIValue(iVal);
+
+  if (!glowConstant) {
+    return;
+  }
+
   addGlowNodeValue(outputs[0], glowConstant->getOutput());
 }
 
 void PyTorchModelLoader::populateNodeLoaderMapping() {
-  nodeLoaderMapping_[at::Symbol::fromQualString("prim::Constant")] =
-      [this](const torch::jit::Node *node) { return loadConstant(node); };
+  nodeLoaderMapping_.emplace(
```

Comment:
This is not the idea way to register these, it's getting a bit unreadable, but it works for now.

Reply:
I think this was changed to a static map in another diff, so you'll need to rebase.

Reply:
I have landed different registration, rebase is required.